app/controllers/s3_harvester_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+##
+# Creating new DLME JSON items by fetching from url
+class S3HarvesterController < Spotlight::ApplicationController
+  before_action :authenticate_user!
+
+  load_and_authorize_resource :exhibit, class: Spotlight::Exhibit
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def create
+    authorize! :create, DlmeJson
+    resp = fetch(params['url'])
+    unless resp&.success?
+      flash[:error] = t('spotlight.resources.fetch.error')
+      return redirect_to spotlight.new_exhibit_resource_path(current_exhibit)
+    end
+    items = resp.body.split("\n")
+    record_count = 0
+    items.each_with_index do |item, index|
+      begin
+        object = DlmeJson.create!(data: { json: item }, exhibit: current_exhibit)
+        object.reindex_later
+      rescue ActiveRecord::RecordInvalid
```

Comment:
Rather than using rescue for flow control, it would be better to do:
```ruby
object = DlmeJson.new(data: ...)
if object.valid?
  # handle invalid
else
  object.save_and_index
end
```

Reply:
Done